### ISSUE TITLE
switch to mariadb

### ISSUE THREAD

**Issue by AndrewGearhart** [open-oni/open-oni#269]
In our shop, we've been progressively moving out of mysql and into mariadb for most things. Since they're binary compatible, it's not usually a significant change. 
**Does anyone have any strong objections?** 
Otherwise, I'll do the work to get us switched.

**Comment by jduss4** [comment]
We use mariadb here, too, so I don't think we have any concerns.  @techgique ?

**Comment by techgique** [comment]
Enterprise Linux (EL) means Red Hat Enterprise Linux or CentOS.

Yeah, we use EL7's MariaDB, which is version 5.5.x. That will of course be supported by Red Hat for the same decade that EL7 is. It looks like the MariaDB project itself is still supporting 5.5 too for anyone not running EL7, such as with Docker perhaps.

I added my [relatively elaborate MySQL config](https://github.com/open-oni/open-oni/blob/master/conf/mysql/openoni.cnf) to open-oni during our last sprint. There were a few changes necessary for use with MariaDB 5.5 on EL7, but I have my config for MariaDB documented as well. It's just not public.

The only external change I noticed when upgrading from EL6's MySQL 5.1 to EL7's MariaDB 5.5 in the past is that float columns show more decimal points if you are using those. Six decimal points instead of two. That can be configured though, IIRC.

So I'm all for switching to MariaDB.
5.5 would be simplest, assuming everyone deploys with EL7.
